pkg/provider/azure_vmss_cache_test.go

```diff
@@ -121,6 +122,20 @@ func TestVMSSVMCache(t *testing.T) {
 	err = ss.DeleteCacheForNode(vmName)
 	assert.NoError(t, err)
 
+	// ensure only one vm is removed from the cache
```

Comment:
thanks adding a test for this scenario!

Reply:
would you add this test into master branch @nilo19 I think master branch does not have this issue, but still needs a ut, thanks.